### PR TITLE
Ft/gf2n: n = 32 & 64

### PR DESCRIPTION
**Pull request by lamphamsy**
# Support GF2N(n=32, 64)

# Operations

| Method | n <= 16 | n > 16 |
| --------| -------- | ------ |
| Multiplication | Log table | Split table |
| Division | Log table | Multiplication + inversion |
| Inversion | Use division | Extended Euclide algorithm |

TODO: division using extended Euclide algorithm

### CODE REVIEW

**Comment by vrancurel** [comment]
It looks good but can you add tests in tools/test_ec.sh ?

File: src/gf2n.h [hunk 1]
```diff
@@ -3,15 +3,39 @@
 
 #include "gf.h"
 
+enum MulType { mul_log_tab, split_8_8 };
+enum DivType { div_log_tab, div_by_inv };
+enum InvType { inv_by_div, inv_ext_gcd };
+
```
Comment (vrancurel) [inline src/gf2n.h]: Can you put enum values in upper case, e.g. MUL_LOG_TAB and inside the class definition ?

File: test/gf_utest.cpp [hunk 2]
```diff
@@ -185,6 +220,7 @@ void gf_utest()
   gfutest_uint32.gf_utest();
   GFUtest<uint64_t> gfutest_uint64;
   gfutest_uint64.gf_utest();
+  gfutest_uint64.gf_utest_2_32();
   GFUtest<mpz_class> gfutest_mpz;
   gfutest_mpz.gf_utest_nogf2n(); //XXX gf2n broken for now
 }
```
Comment (vrancurel) [inline test/gf_utest.cpp]: can you also add a test in tools/test_ec.sh ? e.g. _16 and _32 ?

File: src/gf2n.h (hunk 1 above)
Comment (lamphamsy) [inline src/gf2n.h]: yes sure

File: test/gf_utest.cpp (hunk 2 above)
Comment (lamphamsy) [inline test/gf_utest.cpp]: yes, I forgot that. thanks

**Pull request merged**